include/aikido/robot/ConcreteManipulator.hpp

```diff
@@ -0,0 +1,148 @@
+#ifndef AIKIDO_ROBOT_CONCRETEMANIPULATOR_HPP_
+#define AIKIDO_ROBOT_CONCRETEMANIPULATOR_HPP_
+
+#include "aikido/robot/Hand.hpp"
+#include "aikido/robot/Manipulator.hpp"
+#include "aikido/robot/util.hpp"
+
+namespace aikido {
+namespace robot {
+
+AIKIDO_DECLARE_POINTERS(ConcreteManipulator)
+
+/// A concrete implementation of a manipulator.
+/// ConcreteManipulator supports a collection of
+/// end-effector specific planning methods.
+class ConcreteManipulator : public Manipulator
+{
+public:
+  ConcreteManipulator(RobotPtr robot, HandPtr hand);
+
+  virtual ~ConcreteManipulator() = default;
+
+  // Documentation inherited.
+  virtual HandPtr getHand() override;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> smoothPath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path,
+      const constraint::TestablePtr& constraint) override;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> retimePath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path) override;
+
+  // Documentation inherited.
+  virtual void executeTrajectory(
+      const trajectory::TrajectoryPtr& trajectory) override;
+
+  // Documentation inherited.
+  /// \copydoc Robot::getNamedConfiguration
+  virtual boost::optional<Eigen::VectorXd> getNamedConfiguration(
+      const std::string& name) const override;
+
+  // Documentation inherited.
+  virtual void setNamedConfigurations(
+      std::unordered_map<std::string, const Eigen::VectorXd>
+          namedConfigurations) override;
+
+  // Documentation inherited.
+  virtual std::string getName() const override;
+
+  // Documentation inherited.
+  virtual dart::dynamics::MetaSkeletonPtr getMetaSkeleton() override;
+
+  // Documentation inherited.
+  virtual aikido::statespace::dart::MetaSkeletonStateSpacePtr getStateSpace()
+      override;
+
+  // Documentation inherited.
+  virtual void setRoot(Robot* robot) override;
+
+  // Documentation inherited.
+  virtual void step(
+      const std::chrono::system_clock::time_point& timepoint) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::CollisionFreePtr getSelfCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::TestablePtr getFullCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
```

Comment:
Should this be `statespace::dart::ConstMetaSkeletonStateSpacePtr`?

Reply:
I'll wait for Const PR as this requires in a lot of propagated changes.